ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -53,14 +52,20 @@ public synchronized Bytes32 processHead() {
     StoreTransaction transaction = recentChainData.startStoreTransaction();
     Bytes32 headBlockRoot = protoArrayForkChoiceStrategy.findHead(transaction);
     transaction.commit(() -> {}, "Failed to persist validator vote changes.");
-    BeaconBlock headBlock = recentChainData.getStore().getBlock(headBlockRoot);
-    recentChainData.updateBestBlock(headBlockRoot, headBlock.getSlot());
+    recentChainData.updateBestBlock(
+        headBlockRoot,
+        protoArrayForkChoiceStrategy
+            .blockSlot(headBlockRoot)
+            .orElseThrow(
+                () ->
+                    new IllegalStateException("Unable to retrieve the slot of fork choice head")));
```

Comment:
When we were finding head we already knew the root and slot.  Can we find a way to just return it instead of having to do a second lookup?

Reply:
It could be done but the `findHead` method of ProtoArray is used more than 50 times in tests so I'd be wary to change its API in this PR. Also, I'm not worried since the retrieval is constant time, and `findHead` gets called once (or twice a slot).